lib/matplotlib/cm.py

```diff
@@ -255,10 +254,18 @@ def to_rgba(self, x, alpha=None, bytes=False, norm=True):
                     xx = x
                 else:
                     raise ValueError("third dimension must be 3 or 4")
-                if bytes and xx.dtype != np.uint8:
-                    xx = (xx * 255).astype(np.uint8)
-                if not bytes and xx.dtype == np.uint8:
-                    xx = xx.astype(float) / 255
+                if xx.dtype.kind == 'f':
+                    if xx.max() > 1 or xx.min() < 0:
```

Comment:
As a middle road, I think we can safely assume if `norm=False` is passed in that it's already normalized and we don't need to do this check.  That will catch most of the cases where the check would be redundant.